### ISSUE TITLE
Official support for IDrive E2

### ISSUE THREAD

**Issue by timkgh** [rclone/rclone#6293]
There are some S3 compatible instructions, but I was wondering whether rclone should just support IDrive E2 as an easy config option, with a proper `provider = IDriveE2`, env vars `RCLONE_IDRIVEE2_ACCESS_KEY_ID`, `RCLONE_IDRIVEE2_SECRET_ACCESS_KEY`, etc.

https://www.idrive.com/e2/
https://www.idrive.com/e2/rclone

Related to https://github.com/rclone/rclone/issues/6229, copying working config from there for others who may get tripped up by `server_side_encryption` and `no_check_bucket` config settings:
```
[idrive]
type = s3
provider = IDrive
access_key_id = ACCESS_KEY
secret_access_key = SECRET_KEY
endpoint = ENDPOINT_DOMAIN_FROM_DASHBOARD
acl = private
server_side_encryption = aws:kms
no_check_bucket = true
```

**Comment by animosity22** [comment]
Duplicate of 

https://github.com/rclone/rclone/issues/5229

**Comment by bits01** [comment]
Is it really a duplicate? iDrive has 2 backends: Backup and E2 (S3 compatible). There's no mention of E2 in #5229. I think this issue is still valid to remain open, if the other one is about the _backup_ backend.

**Comment by animosity22** [comment]
Oops. You are correct. My bad!

**Comment by Gigaion** [comment]
In regards to the configuration requirement of 'no_check_bucket = true' & 'server_side_encryption = aws:kms'

This can be simply solved by adding the following yes/no question during the setup prompts:

- Does the provided IDrive access key have All buckets permissions? (y/n)
  - If no: Add the no_check_bucket = true
  - If yes: Don't add no_check_bucket to configuration.

- Does the bucket have 'Default Encryption' enabled on IDrive E2?
  - If no: Don't add server_side_encryption = aws:kms to configuration.
  - If yes: Add the server_side_encryption = aws:kms

Is rclone able to add this logic into the configuration setup, to clear up confusion for future people?

**Comment by timkgh** [comment]
It appears there is official support now in v1.59.0
https://rclone.org/s3/#idrive-e2

**Comment by huyz** [comment]
> Does the provided IDrive access key have All buckets permissions? (y/n)

the `no_check_bucket` prompt was already there when I tried it last week

**Comment by Gigaion** [comment]
> > Does the provided IDrive access key have All buckets permissions? (y/n)
> 
> the `no_check_bucket` prompt was already there when I tried it last week

It exists as an advanced option prompt that doesn't clarify that it's required to be enabled if using a limited access key created by IDrive E2.

By prompting it as part of the non-advanced portion of setup, it clarifies that this is a required configuration step (not an optional advanced configuration).

**Comment by darthShadow** [comment]
Closing since this has been available since 1.59. Any issues with the integration can be opened as new Github Issues.
